nautobot/extras/models/jobs.py

```diff
@@ -799,6 +814,80 @@ def enqueue_job(
         if task_queue is None:
             task_queue = job_model.default_job_queue.name
 
+        job_queue = JobQueue.objects.get(name=task_queue)
+        # Kubernetes Job Queue logic
+        if job_queue.queue_type == JobQueueTypeChoices.TYPE_KUBERNETES:
+            kube_config_context = job_queue.context
+            kube_config_file_path = settings.KUBERNETES_CONFIG_FILE
+            pod_name = settings.KUBERNETES_JOB_POD_NAME
+            pod_namespace = settings.KUBERNETES_JOB_POD_NAMESPACE
+            nautobot_image_name = settings.KUBERNETES_JOB_IMAGE_NAME
+            nautobot_container_name = settings.KUBERNETES_JOB_CONTAINER_NAME
+            nautobot_container_port_number = 8000
+
+            # Load Config file and APIs
+            config.new_client_from_config(
+                config_file=kube_config_file_path, context=kube_config_context, persist_config=False
+            )
+            core_v1 = core_v1_api.CoreV1Api()
+            api_instance = core_v1
+            resp = None
+
+            # Try to read existing pod
+            try:
+                logger.info(f"Reading existing pod {pod_name} in namespace {pod_namespace}")
+                resp = api_instance.read_namespaced_pod(name=pod_name, namespace=pod_namespace)
+                logger.info(f"Found existing pod {pod_name} in namespace {pod_namespace}")
+            except ApiException as e:
+                if e.status != 404:
+                    logger.info(f"Unknown error: {e}")
+
+            # If existing pod does not exist, create a new pod
+            if not resp:
+                logger.info(f"Pod {pod_name} does not exist in namespace {pod_namespace}. Creating it...")
+                pod_manifest = {
+                    "apiVersion": "v1",
+                    "kind": "Pod",
+                    "metadata": {"name": pod_name},
+                    "spec": {
+                        "containers": [
+                            {
+                                "image": nautobot_image_name,
+                                "name": nautobot_container_name,
+                                "ports": [{"containerPort": nautobot_container_port_number}],
```

Comment:
What purpose does port this serve - is it even necessary?

Reply:
Excerpt from https://faun.pub/should-i-configure-the-ports-in-kubernetes-deployment-c6b3817e495:
"containerPortList of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational." I think it is not necessary.